### PR TITLE
Add observer gem as a dependency

### PR DESCRIPTION
**Pull request by AaronRustad**
There is a warning when starting up `retest` that the [observer gem](https://github.com/ruby/observer) will be removed in Ruby 3.4. This PR adds the dependency in.

> /Users/xxx/.asdf/installs/ruby/3.3.0/lib/ruby/gems/3.3.0/gems/retest-1.13.0/lib/retest.rb:4: warning: observer was loaded from the standard library, but will no longer be part of the default gems since Ruby 3.4.0. Add observer to your Gemfile or gemspec. Also contact author of retest-1.13.0 to add observer into its gemspec.

### CODE REVIEW

**Comment by AlexB52** [comment]
Thanks @AaronRustad, the change looks good. 
I'll make a quick research on how this impacts previous ruby version from 2.5+ before we merge the change. 
That said, it might all be fine.

File: retest.gemspec [hunk 1]
```diff
@@ -28,4 +28,5 @@ Gem::Specification.new do |spec|
   spec.add_runtime_dependency "string-similarity", ["~> 2.1"]
   spec.add_runtime_dependency "listen", ["~> 3.9"]
   spec.add_runtime_dependency "tty-option", ["~> 0.1"]
+  spec.add_runtime_dependency "observer", ["~> 0.1.0"]
```
Comment (AlexB52) [inline retest.gemspec:31]: I'm happy to introduce minor version updates from the observer gem. 

```ruby
  spec.add_runtime_dependency "observer", ["~> 0.1"]
```
Can you apply that change and bundle again, please?

I've made a PR that applies the change https://github.com/AaronRustad/retest/pull/1, but feel free to make your own commit on your branch. The PR is mostly to show where all the changes should be applied.

Comment (AaronRustad) [reply retest.gemspec]: Sure thing. Done.

File: Gemfile.lock [hunk 2]
```diff
@@ -3,6 +3,7 @@ PATH
   specs:
     retest (1.13.0)
       listen (~> 3.9)
+      observer (~> 0.1.0)
```
Comment (AlexB52) [inline Gemfile.lock:6]: @AaronRustad 
Sorry, I wasn't clear enough; the minor version should be updated here, too. 
Bundling again locally will probably update the Gemfile.lock accordingly.

Once this is done I will merge and release your changes as `v1.13.1`

Thanks @AaronRustad

Comment (AaronRustad) [reply Gemfile.lock]: Gah. Of course!

**Comment by AlexB52** [comment]
Thank you @AaronRustad, it's been merged and released
https://rubygems.org/gems/retest

**Pull request merged**